packages/components/src/ui/radix-dropdown/index.tsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * External dependencies
+ */
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+import { SVG, Circle } from '@wordpress/primitives';
+import { check, lineSolid } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Icon from '../../icon';
+import * as DropdownMenuStyled from './styles';
+import type {
+	DropdownMenuProps,
+	DropdownSubMenuProps,
+	DropdownItemProps,
+} from './types';
+
+// Observations / Questions:
+// - is it enough on the larger components to have only one forwarded ref?
+//   If we have only one, should it be to the "root", the "trigger", or the "content"?
+// - Should we be consistent in using the same value for the `asChild` prop on:
+//   - trigger
+//   - content
+//   - sub trigger
+//   - sub content
+// - Should we allow customizing the `asChild` prop on trigger and content?
+// - Props & customisability:
+//   - Which props should we expose?
+//   - Should props be "namespaced" for each subcomponent?
+//   - We should probably explicitly `Pick<>` every prop that we want to expose
+// - Subtrigger arrow:
+//   - Should it always be there (ie. an internal implementation)?
+//   - Should we just expect that the the consumers handle it themselves?
+//   - Should we expose it as a separate component that consumers could use?
```

Comment:
It should be either:
- an internal implementation, OR
- if we can do this in a semantivally correct way (via `asChild`), we could just reuse the `suffix` prop

Reply:
@mirka to take a closer look at this

Reply:
Since we're currently using `asChild` for the submenu trigger, I've added the arrow via the standard `suffix` prop on `DropdownMenuItem` (which is used in the `tirgger` prop)

Reply:
This is also solved after adopting Lena's proposed approach https://github.com/WordPress/gutenberg/pull/49473#discussion_r1178259484